subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DependencyLockingProvider.java

```diff
@@ -37,4 +37,6 @@
     RegularFileProperty getLockFile();
 
     ListProperty<String> getIgnoredDependencies();
+
+    void confirmConfigurationNotLocked(String configurationName);
```

Comment:
The contract on this method isn't completely clear. "Confirm" infers a question, but this method doesn't return a boolean. Is it supposed to throw an exception if the confirmation fails? If so, can that be documented?

Reply:
Happy to rename. This method is to be called when a configuration that is not locked has been resolved. So that the locking provider knows it can dump any lock state it may have.
This method does not return anything nor does it throw.